cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -125,61 +200,73 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
             "xsimd_SOURCE AUTO"
   )
 
-  set(ARROW_FOUND TRUE)
-  set(ARROW_LIBRARIES "")
+  # TODO: Will this not get set by rapids_cpm_find?
+  set(ARROW_FOUND
+      TRUE
+      PARENT_SCOPE
+  )
 
-  # Arrow_ADDED: set if CPM downloaded Arrow from Github Arrow_DIR:   set if CPM found Arrow on the
-  # system/conda/etc.
-  if(Arrow_ADDED OR Arrow_DIR)
-    if(BUILD_STATIC)
-      list(APPEND ARROW_LIBRARIES arrow_static)
-    else()
-      list(APPEND ARROW_LIBRARIES arrow_shared)
-    endif()
+  if(BUILD_STATIC)
+    set(ARROW_LIBRARIES
```

Comment:
This isn't doing what you expect. When you set a variable with `PARENT_SCOPE` it becomes set once you return to the parent scope. That means the below code that iterates `ARROW_LIBRARIES` won't work since the local variable `ARROW_LIBRARIES` doesn't exist. 

This breaks builds with arrow built from source. Instead you should use the previous behavior of building a local variable and setting it up to the parent scope

Reply:
I see, I did not realize that. In the case of `ARROW_FOUND` I think the old behavior was actually causing problems because the variable wasn't being propagated up in certain cases, but for `ARROW_LIBRARIES` it makes sense to stick to the older behavior then.